### PR TITLE
Ozone restructuring, Caching of tokens, Unit Tests, Sonar Integration and B3 Propagation opentelemetry

### PR DESCRIPTION
**Pull request by yashmehta1**

### CODE REVIEW

File: configs/config.yaml [hunk 1]
```diff
@@ -2,14 +2,22 @@ log:
...
+bouncer:
+  hydra:
+    url: http://localhost:4445
+    path:
+      introspect: /hydra/oauth2/introspect
+accounts:
+  hydra:
+    url: http://localhost:4445
+    path:
+      introspect: /customer-oauth/oauth2/introspect
```
Comment (ankit-ls) [inline configs/config.yaml:14]: @yashmehta1 Should we rename this to `hydra.bouncer.url` and `hydra.accounts.url`?

File: internal/server/router.go [hunk 2]
```diff
@@ -1,25 +1,40 @@
...
+	"github.com/livspaceeng/ozone/internal/utils"
+	"github.com/patrickmn/go-cache"
 	swaggerfiles "github.com/swaggo/files"
 	ginSwagger "github.com/swaggo/gin-swagger"
+	"go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin"
+)
+
+var (
+	httpClient                                      = &http.Client{}
+	cacheClient										= cache.New(5*time.Minute, 10*time.Minute)
```
Comment (ankit-ls) [inline internal/server/router.go:20]: @yashmehta1 What does this (5*time.Minute, 10*time.Minute) signify?

Comment (ankit-ls) [reply internal/server/router.go:20]: Are we hardcoding cache expiry?

**Review by ankit-ls — COMMENTED** [review]
@yashmehta1 Added few comments, please check them

File: configs/config.yaml (hunk 1 above)
Comment (yashmehta1) [reply configs/config.yaml:14]: Sure @ankit-ls I will make this change

File: internal/server/router.go (hunk 2 above)
Comment (yashmehta1) [reply internal/server/router.go:20]: @ankit-ls Here we're creating a cache with a default expiration time of 5 minutes, and which purges expired items every 10 minutes. Cache expiry is calculated on the basis of hydra response using expiry time. If we set 0 while creating an item in cache, then this default expiration time will be used.

File: configs/config.go [hunk 3]
```diff
@@ -25,7 +26,7 @@ func Init() {
 func setDefaults(config *viper.Viper) {
 	config.SetDefault("log.level", "info")
 	config.SetDefault("server.address", ":32123")
-	config.SetDefault("hydra.url", "localhost:4445")
+	config.SetDefault("bouncer.hydra.url", "localhost:4445")
```
Comment (ankit-ls) [inline configs/config.go:29]: Update this url as well

Comment (yashmehta1) [reply configs/config.go]: Done @ankit-ls

File: internal/server/router.go (hunk 2 above)
Comment (ankit-ls) [reply internal/server/router.go:20]: Cool 👍

**Review by ankit-ls — APPROVED** [review]
Please get it tested by Aman.

**Review by ankit-ls — DISMISSED** [review]
@yashmehta1 Please review it with Aman

**Review by amanLivspace — APPROVED** [review]
@ankit-ls Testing completed

**Review by ankit-ls — APPROVED** [review]

**Pull request merged**